### PR TITLE
upgrade the library to MSL4.0.0 and change version to 2.1.2

### PR DESCRIPTION
**Pull request by adrpo**

### CODE REVIEW

**Review by dietmarw — CHANGES_REQUESTED** [review]
The README and the library itself should also be updated regarding the version number.

**Comment by adrpo** [comment]
> The README and the library itself should also be updated regarding the version number.

Done. Should I maybe change the version to 2.2? I'm a bit unsure.

**Comment by dietmarw** [comment]
Tbh, it should be 3.0.0 but since this lib never followed semver 2.2 may be enough. But then again since nothing has been changed in the lib wrt features, a jump in the minor version might suggest something that is not there. So 2.1.2 might work just as well.

File: SystemDynamics/package.mo [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package SystemDynamics "System Dynamics Library (Version 2.1.1)"
+﻿package SystemDynamics "System Dynamics Library (Version 2.1.1)"
```
Comment (dietmarw) [inline SystemDynamics/package.mo:1]: Here still the old version is present. But it is also a strange place to have the version number. So maybe just remove the version from the package description here.

Comment (adrpo) [reply SystemDynamics/package.mo:1]: Ah, I missed that one. I will remove it as is not needed, the tools display the version.

**Comment by dietmarw** [comment]
Maybe a good idea to also add a version annotation?

**Comment by adrpo** [comment]
> Maybe a good idea to also add a version annotation?

There is one annotation `version="2.1.2", uses(Modelica(version="4.0.0")),` buried alive between the graphical annotations.

**Comment by dietmarw** [comment]
Ah, got to love a massive annotation that is spanning over just one line. ;-)

**Review by dietmarw — APPROVED** [review]
Looks good now.

**Comment by adrpo** [comment]
I can move the version annotation at the end or the beginning.

**Comment by dietmarw** [comment]
Nah. it's fine. I don't care.

**Pull request merged**